ush/setup.py

```diff
@@ -701,62 +698,43 @@ def get_location(xcs, fmt, expt_cfg):
-    if workflow_config["CCPP_PHYS_SUITE"] in hires_ccpp_suites:
-        dt = fcst_config.get("DT_ATMOS")
-        if dt:
-            if dt > 40:
-                logger.warning(dedent(
-                    f"""
-                    WARNING: CCPP suite {workflow_config["CCPP_PHYS_SUITE"]} requires short
-                    time step regardless of grid resolution. The user-specified value
-                    DT_ATMOS = {fcst_config.get("DT_ATMOS")}
-                    may result in CFL violations or other errors!
-                    """
-                ))
 
     # Gather the pre-defined grid parameters, if needed
-    if workflow_config.get("PREDEF_GRID_NAME"):
-        grid_params = set_predef_grid_params(
-            USHdir,
-            workflow_config["PREDEF_GRID_NAME"],
-            fcst_config["QUILTING"],
-        )
+    if grid_name := workflow_config.get("PREDEF_GRID_NAME"):
+        grid_params = get_yaml_config(Path(USHdir, "predef_grid_params.yaml"))[
+            grid_name
+        ]
 
         # Users like to change these variables, so don't overwrite them
         special_vars = ["DT_ATMOS", "LAYOUT_X", "LAYOUT_Y", "BLOCKSIZE"]
-        for param, value in grid_params.items():
-            if param in special_vars:
-                param_val = fcst_config.get(param)
-                if param_val and isinstance(param_val, str) and "{{" not in param_val:
-                    continue
-                elif isinstance(param_val, (int, float)):
-                    continue
-                # DT_ATMOS needs special treatment based on CCPP suite
-                elif param == "DT_ATMOS":
-                    if workflow_config["CCPP_PHYS_SUITE"] in hires_ccpp_suites and grid_params[param] > 40:
-                        logger.warning(dedent(
-                            f"""
-                            WARNING: CCPP suite {workflow_config["CCPP_PHYS_SUITE"]} requires short
-                            time step regardless of grid resolution; setting DT_ATMOS to 40.\n
-                            This value can be overwritten in the user config file.
-                            """
-                        ))
-                        fcst_config[param] = 40
-                    else:
-                        fcst_config[param] = value
-                else:
-                    fcst_config[param] = value
-            elif param.startswith("WRTCMP"):
-                if fcst_config.get(param) == "":
-                    fcst_config[param] = value
-            elif param == "GRID_GEN_METHOD":
-                workflow_config[param] = value
-            else:
-                grid_config[param] = value
+        for var in special_vars:
+            user_value = fcst_config[var]
+            if isinstance(user_value, int):
+                # Don't keep the pre-defined value
+                del grid_params["task_run_fcst"][var]
+        expt_config.update_from(grid_params)
```

Comment:
This is the reduction in logic we get when mapping the grid YAML file to the same structure as the config_defaults file.